packages/ts/react-signals/src/NumberSignal.ts

```diff
@@ -38,24 +38,28 @@ export class NumberSignal extends ValueSignal<number> {
    *
    * @param delta - The delta to increment the value by. The delta can be
    * negative.
+   * @returns An operation object that allows to perform additional actions.
    */
-  incrementBy(delta: number): void {
+  incrementBy(delta: number): Operation {
     if (delta === 0) {
-      return;
+      return noOperation;
```

Comment:
I'm a bit confused why a `noOperation` should be returned in this case?
If I understand the `noOperation` correctly, the then callback that is passed to a `noOperation` is not executed. 
I would suggest to change this to an implementation of `Operation` interface, that calls the passed-in callback immediately without storing anything in the `thenCallbacks` map, so that something like the following example manages to call the `notifySubscribers` function after the `incrementBy` is done, even in case the result of `calculateAmount()` happened to be `0`:
```
let amount = 1;
if (some condition is satisfied) {
    amount = calculateAmount();
}
numberSignal.incrementBy(amount).result.then(() => notifySubscribers());
```

Reply:
Or maybe I'm not understanding the `noOperation` implementation correctly.

Reply:
It depends on what should happen when 0 is passed as value, or any other "void" value for the other signal operations. It was my understanding that the callback must be run when an update happens, so as nothing changed on the server (it returns immediately without processing), I see no reason to process any callback and, in your example, notify the subscribers.

Reply:
Well the user might not know about all the possible outcomes of `calculateAmount` that can happen at runtime, so it will be smoother for them to just chain the `notifySubscribers()` to be done after the `incrementBy`. To me, an `incrementBy(0)` is a successful operation without the need to make the roundtrip to the server.